### PR TITLE
KRBN-6777: Add pulse counters required by addons

### PR DESCRIPTION
**Pull request by johnliu25**
Add pulse counters required by addons like csi and cnds

### CODE REVIEW

File: karbon/karbon_cluster_registration_structs.go [hunk 1]
```diff
@@ -169,3 +161,22 @@ type K8sUpdateClusterRegistrationAddonMetricsResponse struct {
 	// cluster uuid
 	ClusterUUID string `json:"cluster_uuid,omitempty"`
 }
+
+// K8sClusterMetrics Metrics information for this k8s cluster.
+type K8sClusterMetrics map[string]K8sClusterResourceList
+
+type K8sClusterResourceList []*K8sClusterResource
```
Comment (thunderboltsid) [inline karbon/karbon_cluster_registration_structs.go]: add a comment on all exported types

File: karbon/karbon_cluster_registration_structs.go [hunk 2]
```diff
@@ -169,3 +161,22 @@ type K8sUpdateClusterRegistrationAddonMetricsResponse struct {
...
+
+type K8sClusterAddonMetrics map[string]K8sClusterResourceList
```
Comment (thunderboltsid) [inline karbon/karbon_cluster_registration_structs.go:168]: ditto

File: karbon/karbon_cluster_registration_structs.go [hunk 3]
```diff
@@ -169,3 +161,22 @@ type K8sUpdateClusterRegistrationAddonMetricsResponse struct {
...
+
+// K8sClusterResource k8s cluster resource
```
Comment (thunderboltsid) [inline karbon/karbon_cluster_registration_structs.go:172]: Comment needs to be of the format
// Noun verb ...

File: karbon/karbon_cluster_registration_service.go [hunk 4]
```diff
@@ -105,3 +105,31 @@ func (op ClusterRegistrationOperations) UpdateK8sRegistrationAddonInfo(ctx conte
 	}
 	return karbonClusterActionResponse, nil
 }
+
+// UpdateK8sRegistrationInfo updates k8s info
+func (op ClusterRegistrationOperations) UpdateK8sRegistrationMetrics(ctx context.Context, k8sClusterUUID string, updateInfoRequest *K8sUpdateClusterRegistrationMetricsRequest) (*K8sUpdateClusterRegistrationMetricsResponse, error) {
```
Comment (thunderboltsid) [inline karbon/karbon_cluster_registration_service.go:110]: what do you think about the k8sClusterUUID being a typed UUID instead?

**Review by deepakm-ntnx — DISMISSED** [review]

**Review by thunderboltsid — APPROVED** [review]

**Review by deepakm-ntnx — APPROVED** [review]

**Pull request merged**